internal/authenticateflow/stateful.go

```
@@ -179,6 +193,281 @@ func (s *Stateful) SignIn(
 	return nil
 }
 
+func getSbrID(r *http.Request, state *sessions.Handle) string {
+	query := r.URL.Query()
+	if query.Has(urlutil.QueryBindSession) {
+		sbrID := query.Get(urlutil.QueryBindSession)
+		if state == nil {
+			panic("bug: state missing")
+		}
```

I'm a little confused by this. It looks like the `state` parameter is otherwise unused in this function? Is this more of a "validateRequest" method?

I'm also a little nervous around adding `panic`s. Would it be better to return an error from the methods that require a non-nil `state`?

---

Same as blow, I think a bad rebase re-introduced this logic, the state / redirect can/is handled by the authenticate middleware here, so I've removed this